selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/workspace/TestWorkspaceInjector.java

```diff
@@ -11,57 +11,29 @@
  */
 package org.eclipse.che.selenium.core.workspace;
 
-import static com.google.common.base.Strings.isNullOrEmpty;
-import static java.lang.String.format;
-
 import com.google.inject.Injector;
-import com.google.inject.Key;
-import com.google.inject.MembersInjector;
 import com.google.inject.Provider;
-import com.google.inject.name.Names;
 import java.lang.reflect.Field;
-import org.eclipse.che.selenium.core.user.DefaultTestUser;
 
 /**
  * Injector for custom annotation {@link InjectTestWorkspace}.
  *
  * @author Anatolii Bazko
  */
-public class TestWorkspaceInjector<T> implements MembersInjector<T> {
-  private final Field field;
-  private final InjectTestWorkspace injectTestWorkspace;
-  private final Provider<Injector> injectorProvider;
+public class TestWorkspaceInjector<T> extends AbstractTestWorkspaceInjector<T> {
 
   public TestWorkspaceInjector(
       Field field, InjectTestWorkspace injectTestWorkspace, Provider<Injector> injectorProvider) {
-    this.field = field;
-    this.injectTestWorkspace = injectTestWorkspace;
-    this.injectorProvider = injectorProvider;
+    super(field, injectTestWorkspace, injectorProvider);
   }
 
   @Override
   public void injectMembers(T instance) {
-    Injector injector = injectorProvider.get();
-
-    TestWorkspaceProvider testWorkspaceProvider = injector.getInstance(TestWorkspaceProvider.class);
-    int workspaceDefaultMemoryGb =
-        injector.getInstance(Key.get(int.class, Names.named("workspace.default_memory_gb")));
-
     try {
-      DefaultTestUser testUser =
-          isNullOrEmpty(injectTestWorkspace.user())
-              ? injector.getInstance(DefaultTestUser.class)
-              : findInjectedUser(instance, injectTestWorkspace.user());
-
-      int memoryGb =
-          injectTestWorkspace.memoryGb() <= 0
-              ? workspaceDefaultMemoryGb
-              : injectTestWorkspace.memoryGb();
-
       TestWorkspace testWorkspace =
```

Comment:
According the changes [here](https://github.com/Katka92/rh-che/commit/f36760ebad39a9a925a8de1053e6bef42776e26a)  the main difference is how workspace is created, if you replace `RhCheTestWorkspaceImpl` by `TestWorkspace`.
If so, I would propose to make just this code part of code an abstract:
```
AbstractTestWorkspaceInjector.java
...
  injectMembers(T instance) {
    ...
     TestWorkspace testWorkspace = createWorkspace();
      testWorkspace.await();
      field.setAccessible(true);
      field.set(instance, testWorkspace);
     ....
   }

  abstract TestWorkspace createWorkspace();
...

}